src/PhpWord/TemplateProcessor.php

```diff
@@ -315,6 +318,123 @@ public function setComplexBlock($search, Element\AbstractElement $complexType):
         $this->replaceXmlBlock($search, $xmlWriter->getData(), 'w:p');
     }
 
+    /**
+     * @param string $search
+     * @param string $htmlContent
+     * @param bool $fullHtml
+     */
+    public function setHtmlBlock($search, $htmlContent, $fullHtml = false): void
+    {
+        $phpWord = new PhpWord();
+        $section = $phpWord->addSection();
+        //deal remote load Image
+        $pattern = '/<img[^>]+src\s*=\s*["\']([^"\']+)["\'][^>]*>/i';
+        preg_match_all($pattern, $htmlContent, $matches);
+        $imageSrcList = $matches[1];
+        if (!empty($imageSrcList)) {
+            foreach ($imageSrcList as $imageSrc) {
+                try {
+                    $content = file_get_contents($imageSrc);
```

Comment:
This section of the code is quite clever, but (a) I'm not really sure it's needed, and (b) I don't think you've coded it correctly. I would personally eliminate all the code after `addSection` and before `addHtml` and just let it fail later if the file isn't available; your changes to your test member are sufficient to address my original concern by making the files local rather than external. But, I can see that this checking might be perceived as a benefit, so let's discuss (b). `file_get_contents` does not normally throw an exception (I am not expert in the Php internals, so I suppose there might be some edge case where it throws); it normally just returns `false` if it fails and issues some warning messages describing the failure. So, I think what you want to do is:
```php
foreach ... {
    $content = @file_get_contents(...); // suppress warning messages
    if ($content === false) {
        $localImg = ...
        $htmlContent = ...
    }
}
```